models/userModel.js

```diff
@@ -0,0 +1,89 @@
+const _ = require('lodash');
+const jwt = require('jsonwebtoken');
+const crypto = require('crypto-js');
+const config = require('config');
+const { User } = require('database').models;
+
+const destroyLastSession = async ({ user }) => {
+  if (_.get(user, 'auth.sessions', false) && user.auth.sessions.length > config.limit_sessions) {
+    await User.updateOne({ _id: user._id }, { $pull: { 'auth.sessions': { _id: user.auth.sessions[0]._id } } });
+  }
+};
+
+const destroySession = async ({ userId, session }) => {
+  await User.updateOne({ _id: userId }, { $pull: { 'auth.sessions': { _id: session._id } } });
+};
+
+const findUserBySocial = async ({ id, provider }) => User.findOne({ 'auth.provider': provider, 'auth.id': id }).lean();
+
+const findUserById = async (id) => {
+  try {
+    return { user: await User.findOne({ _id: id }).lean() };
+  } catch (error) {
+    return { error };
+  }
+};
```

Comment:
this two method are duplicated, you can make it universal just by passing a parameter condition and calling findOne with this condition